tabular/src/autogluon/tabular/configs/config_helper.py

```diff
@@ -0,0 +1,357 @@
+from __future__ import annotations
+
+import copy
+from typing import Union
+
+from sklearn.base import BaseEstimator
+
+from autogluon.features import AutoMLPipelineFeatureGenerator
+from autogluon.tabular.configs.hyperparameter_configs import hyperparameter_config_dict
+from autogluon.tabular.configs.presets_configs import tabular_presets_dict
+from autogluon.tabular.trainer.model_presets.presets import MODEL_TYPES
+
+
+class FeatureGeneratorBuilder:
+    def __init__(self, parent):
+        self.parent = parent
```

Comment:
Why does this need a parent? A FeatureGenerator is useful outside of a Predictor. What if I want to just build a FeatureGenerator without a Predictor?

Reply:
To return control to the main builder after it is done:
```python
config = (ConfigBuilder()  
          .included_model_types(['GBM', 'KNN'])
          .feature_generator()   # <<<< switching to FeatureGeneratorBuilder
                    .enable_numeric_features(False).enable_text_ngram_features(False)
                    .build()  # <<<< Return control back to ConfigBuilder
          .time_limit(60)
          .build())
```

Reply:
I understand, I mean what if someone doesn't want to make a TabularPredictor config and instead just a FeatureGenerator config?

Reply:
Updated the code to handle this case; parent now is optional: if passed, then it will pass control back; if `None`, then it will return the generator back.
```
FeatureGeneratorBuilder().enable_text_special_features(False).enable_vision_features(False).build()
```